### PR TITLE
WIP+RF: make serve_path serve from any dir (Closes #138)

### LINKED ISSUE #138: modify serve_path_via_http (in utils.py) to serve any path (not only those from the cur dir)
atm using Thread; make it use multiprocess'ing and allow to change directory

### PR DESCRIPTION
**Pull request by jgors**

### CODE REVIEW

File: datalad/tests/utils.py [hunk 1]
```diff
@@ -222,39 +223,76 @@ def log_message(self, format, *args):
         lgr.debug("HTTP: " + format % args)
 
 
-def serve_path_via_http():
+#def _multiproc_serve_path_via_http(port, path_to_serve_from):
+    #cmd = 'python -m SimpleHTTPServer {}'.format(port)
```
Comment (yarikoptic) [inline datalad/tests/utils.py]: in the end -- remove these commented out attempts

File: datalad/tests/utils.py [hunk 2]
```diff
@@ -222,39 +223,76 @@ def log_message(self, format, *args):
...
+            #if ret_val:
+                #continue
+            #else:
+                #sock.shutdown(1)
+                #sock.close()
+                #break
+
+        multi_proc = multiprocessing.Process(target=_multiproc_serve_path_via_http, args=(port, path))
+        #multi_proc = multiprocessing.Process(target=_multiproc_serve_path_via_http, args=(hostname, path))
+        multi_proc.start()
```
Comment (yarikoptic) [inline datalad/tests/utils.py]: so what about bringing back that looping to get a port which works?

**Comment by yarikoptic** [comment]
@jgors  You would need to merge current master in and resolve conflicts before travis/buildbot could start testing it again.

**Comment by yarikoptic** [comment]
merged, and tuned to skip unicode whenever it can't be supported

**Comment by bpoldrack** [comment]
I'd just like to mention, that "WIP"-marker should disappear before merge. To me, that's the point to have a final look at it. Just to keep in mind next time ;)

**Comment by yarikoptic** [comment]
Yeah... I just was in a bad mood :-)

**Pull request merged**